src/legacy/ui/public/vis/editors/default/vis_options_props.tsx

```diff
@@ -20,7 +20,11 @@
 import { Vis, VisParams } from 'ui/vis';
 
 export type VisOptionsSetValue = (paramName: string, value: unknown) => void;
-export interface VisOptionsProps {
+
+export interface InjectedDependencies {
+  [key: string]: any;
+}
+export interface VisOptionsProps extends InjectedDependencies {
```

Comment:
I wouldn't extend `VisOptionsProps` itself because it will influence other components (`Pie`, `Tag cloud`) allowing to pass almost any addition props. Instead of this let's create a separate interface for tile and region maps (e.g. `VisMapOptionsProps`) which will extend `VisOptionsProps` interface and have `serviceSettings` property additionally.

Reply:
Nice catch! Done!